library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -182,16 +189,16 @@ public void run() {
 
     private void runInternal() {
         // Skip when download already marked as finished; this download was probably started again while racing with UpdateThread.
-        int downloadStatus = FileDownloadInfo.queryDownloadStatus(getContentResolver(), originalDownloadInfo.getId());
-        if (downloadStatus == Downloads.Impl.STATUS_SUCCESS) {
+        int downloadStatus = FileDownloadInfo.queryDownloadStatus(getContentResolver(), originalDownloadInfo.getId(), downloadsUriProvider);
```

Comment:
I guess that is downloadStatus and the class should be DownloadStatus instead of DownloadsStatus